.github/workflows/build.yaml

```diff
@@ -32,12 +33,39 @@ jobs:
         username: ${{ github.actor }}
         password: ${{ secrets.GITHUB_TOKEN }}
 
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+      with:
+        platforms: linux/amd64
+        driver-opts: |
+          image=moby/buildkit:buildx-stable-1
+          network=host
+
+    - name: Cache Docker layers
+      uses: actions/cache@v4
+      with:
+        path: |
+          ${{ env.DOCKER_CACHE_PATH }}
+        key: docker-amd64-${{ env.BRANCH_NAME }}-${{ env.SHORT_REV }}
+        restore-keys: |
+          docker-amd64-${{ env.BRANCH_NAME }}-
+          docker-amd64-jobrad-14.0-
+          docker-amd64-
+
     - name: Build and push Docker image (AMD64)
-      shell: bash
-      run: |
-        IMAGE="${GHCR_IMAGE_NAME}:${SHORT_REV}-amd64"
-        docker build . --platform linux/amd64 -f docker/Dockerfile -t "${IMAGE}" --pull
-        docker push "${IMAGE}"
+      uses: docker/build-push-action@v6
+      with:
+        context: .
+        file: docker/Dockerfile
+        platforms: linux/amd64
+        push: true
+        tags: |
+          ${{ env.GHCR_IMAGE_NAME }}:${{ env.SHORT_REV }}-amd64
+        cache-from: |
+          type=local,src=${{ env.DOCKER_CACHE_PATH }}
+        cache-to: |
+          type=local,dest=${{ env.DOCKER_CACHE_PATH }}
```

Comment:
You are using local cache together with the "cache" action.

1. Wouldn't it be more straightforward to use the "gha" cache type?
2. If we stick with cache type "local", do we need to be mindful of disk space on the machine where the Github runner is running?